internal/dataservice/server.go

```diff
@@ -929,12 +943,45 @@ func (s *Server) SaveBinlogPaths(ctx context.Context, req *datapb.SaveBinlogPath
 	for k, v := range ddlMeta {
 		meta[k] = v
 	}
+	// Save into k-v store
 	err = s.SaveBinLogMetaTxn(meta)
 	if err != nil {
 		resp.Reason = err.Error()
 		return resp, err
 	}
+	// write flush msg into segmentInfo/flush stream
+	msgPack := composeSegmentFlushMsgPack(req.SegmentID)
+	err = s.flushMsgStream.Produce(&msgPack)
+	if err != nil {
+		s.RemoveBinLogMetaTxn(meta) // removes bin log meta from store, like a rollback
+		resp.Reason = err.Error()
+		return resp, err
+	}
```

Comment:
I don't think this's a good solution.

Saving paths into etcd is a valid flush-completed signal. Even if we fail to publish this message, we don't need to redo the whole `consume -> buffer -> flush-> save to etcd -> publish` procedure again. We only have to re-publish this message.

Reply:
The recovery of publishing messages into pulsar can be accomplished in the next PR, this PR do things right if not crashed. So I approve of this change.